reporter/src/main/kotlin/reporters/CycloneDxReporter.kt

```diff
@@ -167,13 +168,15 @@ class CycloneDxReporter : Reporter {
     }
 
     private fun addPackageToBom(input: ReporterInput, pkg: Package, bom: Bom) {
-        // TODO: We should actually use the concluded license expression here, but we first need a workflow to
-        //       ensure it is being set.
         val resolvedLicenseInfo = input.licenseInfoResolver.resolveLicenseInfo(pkg.id)
+            .filter(LicenseView.CONCLUDED_OR_REST)
```

Comment:
@sschuberth I'm not sure if I understand the `TODO` correctly, would this address it or was it meant differently?

Reply:
I believe this *almost* addresses to `TODO`: We should only add a "concluded license" below if we *really* have an explicit concluded license, and not fall back to the union of declared and detected licenses, as those already have their dedicated "declared license" and "detected license" entries.

Reply:
If there is a concluded license, only licenses contained in this license will be added below, but if they are also declared or detected they will be added respectively. For example:
```
concludedLicense = "Apache-2.0 AND MIT"
declaredLicense = "Apache-2.0 AND MPL-1.0"
detectedLicense = "MIT AND BSD-2-Clause"
```
will add "concluded license" and "declared license" entries for "Apache-2.0" and "concluded license" and "detected license" for "MIT", but no entries for "MPL-1.0" and "BSD-2-Clause". Would you prefer that only the "concluded license" entries are created in this case?

Reply:
That's not what I meant, but what you're describing is also not what I'd expect. For your example, I'd expect the CycloneDX BOM to contain the "unfiltered" licenses, i.e. the following license origins:

"concluded license" -> "Apache-2.0"
"concluded license" -> "MIT"
"declared license" -> "Apache-2.0"
"declared license" -> "MPL-1.0"
"detected license" -> "MIT"
"detected license" -> "BSD-2-Clause"

I'm surprised that the concluded license seems to have an effect on the declared and detected licenses. Is the new API designed so that from declared and detected licenses only those expressions are kept that are also part of the concluded license?

Anyway, what I meant originally was that if there's no concluded license, then there should be no "concluded license" origin entry in the BOM. But after looking at the implementation of  `mapLicenseNamesToObjects` again I see that's already the case.

Reply:
> I'm surprised that the concluded license seems to have an effect on the declared and detected licenses. Is the new API designed so that from declared and detected licenses only those expressions are kept that are also part of the concluded license?

That's only because I used `LicenseView.CONCLUDED_OR_REST` which seemed consistent with other reports. But to get what you expect I have removed this filter, it should now be correct.

Reply:
> But to get what you expect

I was considering my expectation to be wrong, but in general I believe declared and detected licenses should be immutable and never be tampered with. If we need some kind of processing of these, we should create new properties, like `declaredLicensesProcessed`, with a different name to make clear these are not the original declared / detected licenses. Would you agree?

> I have removed this filter, it should now be correct.

Thanks, I'll have a look.

Reply:
> I was considering my expectation to be wrong, but in general I believe declared and detected licenses should be immutable and never be tampered with. If we need some kind of processing of these, we should create new properties, like `declaredLicensesProcessed`, with a different name to make clear these are not the original declared / detected licenses. Would you agree?

I think that's missing the point, because `LicenseView` is simply a filter on the license data that can be used on demand. It's been used in the evaluator for a long time, and for example you implemented the logic of `CONCLUDED_OR_REST` in the Antenna reporter.

Reply:
> you implemented the logic of `CONCLUDED_OR_REST` in the Antenna reporter.

Well, I'm using the concluded license if there is one, and otherwise the *unmodified* declared and detected licenses. If that's also the logic of `LicenseView` I'm all fine with that.

However, if there *is* a concluded license, and you *still* query the declared or detected license, in my view the latter two should *not* be modified compared to if there was no concluded license (or we should make this behavior *very* clear and explicit in the API, by naming properties accorindgly). Your previous example gave me the impression that exactly this "filtering" of declared or detected licenses takes place if there is a concluded license.

Reply:
I see, the reason for that behavior is that the `LicenseView` removes licenses which are not part of the concluded license, but keeps all sources for the licenses which are kept, see:
https://github.com/oss-review-toolkit/ort/blob/5d86c10e869ae888b7caaa9c509b3e3cb4d4c43d/model/src/main/kotlin/licenses/LicenseView.kt#L110

Reply:
Right, and I believe that implementation needs to be discussed.